src/S3/ObjectCopier.php

```diff
@@ -74,8 +77,15 @@ public function __construct(
     public function promise()
     {
         return \GuzzleHttp\Promise\coroutine(function () {
+            $headObjectCommand = $this->client->getCommand(
+                'HeadObject',
+                $this->options['params'] + $this->source
+            );
+            if (is_callable($this->options['before_lookup'])) {
```

Comment:
Yes, but only to fulfill what was previously defined and handled in a broken and inconsistent manner. See #1232